projects/packages/jetpack-mu-wpcom/bin/sync-newspack-blocks.sh

```diff
@@ -127,31 +129,95 @@ mkdir -p $TARGET/components
 # beginning with a space to avoid matching it as a substring. (Not perfect, but
 # imperfections will be caught by CI with failing tsc, etc.)
 sed "${sedi[@]}" -e "s| function| Function|g" "$TARGET/types/index.d.ts"
 
 # Note: I would have used eslint-nibble, but it doesn't support autofixing via the CLI.
 echo "Changing JS textdomain to match jetpack-mu-wpcom..."
-pnpm --package=eslint@8.57.0 dlx eslint --no-ignore --rule '"@wordpress/i18n-text-domain":["error",{"allowedTextDomain":"jetpack-mu-wpcom"}]' --fix $TARGET > /dev/null
+BASE=$(cd "$(dirname "${BASH_SOURCE[0]}")"/../../../.. && pwd)
+FULLTARGET="$PWD/$TARGET"
+
+# Add a temporary single-rule eslint.config.mjs file.
+cat > "$TARGET/eslint.config.mjs" <<EOF
+import makeBaseConfig from 'jetpack-js-tools/eslintrc/base.mjs';
+
+// This directory is copy-pasted from elsewhere, but we still need to run this one rule over it so we can't just .eslintignore.
+export default [
+	// Import base config, but no rules.
+	...makeBaseConfig( import.meta.url ).map( block => ( { ...block, rules: {} } ) ),
+	// Enable just this one rule.
+	{
+		rules: {
+			"@wordpress/i18n-text-domain": [ "error", { allowedTextDomain: "jetpack-mu-wpcom" } ],
+		}
+	},
+];
+EOF
+( cd "$BASE" && pnpm run lint-file --no-inline-config --no-ignore --fix "$FULLTARGET" )
+rm "$TARGET/eslint.config.mjs"
 
 echo "Changing JS translation function call to avoid bad minification..."
 pnpm --package=jscodeshift dlx jscodeshift -t ./bin/sync-newspack-blocks-formatter.js --extensions=js $TARGET
 
+# Add temporary PHPCS config file.
+PHPCSSTANDARDFILE="$TARGET/phpcs.tmp.xml"
+cat > "$PHPCSSTANDARDFILE" <<EOF
+<?xml version="1.0"?>
+<ruleset>
+	<rule ref="Jetpack.Functions.I18n">
+		<properties>
+			<property name="text_domain" value="jetpack-mu-wpcom" />
+		</properties>
+	</rule>
+
+	<rule ref="WordPress.Utils.I18nTextDomainFixer">
+		<properties>
+			<property name="old_text_domain" type="array">
+				<element value="newspack-blocks" />
+			</property>
+			<property name="new_text_domain" value="jetpack-mu-wpcom" />
+		</properties>
+	</rule>
+</ruleset>
+EOF
 echo "Changing PHP textdomain to match jetpack-mu-wpcom..."
-../../../vendor/bin/phpcbf --standard=./.phpcs.dir.xml --filter=../../../vendor/automattic/jetpack-phpcs-filter/src/PhpcsFilter.php --runtime-set jetpack-filter-no-ignore -q $TARGET
-
+"$BASE"/vendor/bin/phpcbf --standard="$PHPCSSTANDARDFILE" "$TARGET"
+rm "$PHPCSSTANDARDFILE"
+
+# Add textdomain to block.json
+echo "Adding textdomain to all block.json files..."
+for block_json_file in "$TARGET"/blocks/*/block.json; do
+	TMPFILE=$(mktemp)
+	jq --tab '. += {"textdomain": "jetpack-mu-wpcom"}' "$block_json_file" > "$TMPFILE"
+	mv "$TMPFILE" "$block_json_file"
+done
+
+echo
+
+# Warn about the need to use ENT_COMPAT.
+echo "Please ensure htmlentities and html_entity_decode use 'ENT_COMPAT'!"
+ent_compat_needed=$(grep -rino 'html_entity_decode\|htmlentities' --include="$TARGET/*.php")
```

Comment:
This doesn't seem to work.
```suggestion
ent_compat_needed=$(grep -rino 'html_entity_decode\|htmlentities' --include="*.php" "$TARGET")
```

Reply:
Works for me:

```
--snip--

Please ensure htmlentities and html_entity_decode use 'ENT_COMPAT'!
Detected the below instances:
./src/features/newspack-blocks/synced-newspack-blocks/class-newspack-blocks.php:1104:html_entity_decode
./src/features/newspack-blocks/synced-newspack-blocks/class-newspack-blocks.php:1104:htmlentities

Sync done.
```

That said, it'll only detect files directly in said folder.

Either way, it should soon be a non-issue.

Reply:
Upon further thought, no need to wait to remove this: bc7ac12621